src/networking.c

```diff
@@ -406,19 +406,23 @@ void addReplyError(client *c, const char *err) {
     addReplyErrorLength(c,err,strlen(err));
 }
 
-void addReplyErrorFormat(client *c, const char *fmt, ...) {
-    size_t l, j;
-    va_list ap;
-    va_start(ap,fmt);
-    sds s = sdscatvprintf(sdsempty(),fmt,ap);
-    va_end(ap);
+void addReplyErrorSds(client *c, sds s) {
```

Comment:
Hi @oranagra, addReplySds() take ownership of the SDS allocation. I think it's confusing to add this API that does things differently. In general I believe that the "SDS" API should be only low level, where you give it a piece of protocol, in order to compensate all the cases that the API can't handle, but that's all. The other stuff should be handled differently.

Reply:
@antirez i'm still not sure what's the best pick here.
I renamed it to addReplyErrorSafe which takes char* and length.
let me know if you have a better idea.
please have a look at the new commit with regards to all your comments.
thanks.